ftl/php/builder.py

```diff
@@ -33,27 +34,38 @@ def __init__(self, ctx, args, cache_version_str):
         super(PHP, self).__init__(ctx, _PHP_NAMESPACE, args, cache_version_str,
                                   [_COMPOSER_LOCK, _COMPOSER_JSON])
 
+    def _parse_composer_pkgs(self):
+        descriptor_contents = ftl_util.descriptor_parser(
+            self._descriptor_files, self._ctx)
+        composer_json = json.loads(descriptor_contents)
+        pkgs = []
+        for k, v in composer_json['require'].iteritems():
+            pkgs.append(k + ' ' + v)
+        return pkgs
+
     def Build(self):
         lyr_imgs = []
         lyr_imgs.append(self._base)
         if ftl_util.has_pkg_descriptor(self._descriptor_files, self._ctx):
-            pkg = self.PackageLayer(self._ctx, self._descriptor_files, None,
-                                    self._args.destination_path)
-            cached_pkg_img = None
-            if self._args.cache:
-                with ftl_util.Timing("checking cached pkg layer"):
-                    cached_pkg_img = self._cash.GetAndCheckTTL(
-                        self._base, self._namespace, pkg.GetCacheKey())
-            if cached_pkg_img is not None:
-                pkg.SetImage(cached_pkg_img)
-            else:
-                with ftl_util.Timing("building pkg layer"):
-                    pkg.BuildLayer()
+            pkgs = self._parse_composer_pkgs()
+            for pkg_txt in pkgs:
+                pkg = self.PackageLayer(self._ctx, self._descriptor_files,
+                                        pkg_txt, self._args.destination_path)
+                cached_pkg_img = None
                 if self._args.cache:
-                    with ftl_util.Timing("uploading pkg layer"):
-                        self._cash.Store(self._base, self._namespace,
-                                         pkg.GetCacheKey(), pkg.GetImage())
-            lyr_imgs.append(pkg)
+                    with ftl_util.Timing("checking cached pkg layer"):
```

Comment:
nit: i think you used underscores in here in other places?

Reply:
i think I may have undid this in all places, so this is consistent currently.  Should I change all of these prints to be underscored?

Reply:
nah, it's fine as long as it's all consistent.